app/models/billable_metric.rb

```diff
@@ -20,9 +20,13 @@ class BillableMetric < ApplicationRecord
     max_agg
     unique_count_agg
     recurring_count_agg
+    weighted_sum_agg
   ].freeze
 
+  WEIGHTED_INTERVAL = { seconds: 'seconds' }.freeze
+
   enum aggregation_type: AGGREGATION_TYPES
+  enum weighted_interval: WEIGHTED_INTERVAL
 
   validate :validate_recurring
```

Comment:
Is this new BM compatible with recurring? If no, we should update this validation

Reply:
Yes, it will allow recurring as well